src/language-js/needs-parens.js

```diff
@@ -110,6 +110,37 @@ function needsParens(path, options) {
       }
     }
 
+    // `(type) satisfies never;` and similar cases
+    if (key === "expression") {
+      const ancestorNeitherAsNorSatisfies = path.findAncestor((node) => {
+        switch (node.type) {
+          case "TSSatisfiesExpression":
+          case "SatisfiesExpression":
+          case "TSAsExpression":
+          case "AsExpression":
+          case "AsConstExpression":
+            return false;
+          default:
+            return true;
+        }
+      });
+      if (
+        ancestorNeitherAsNorSatisfies !== parent &&
+        ancestorNeitherAsNorSatisfies.type === "ExpressionStatement"
+      ) {
+        switch (node.name) {
```

Comment:
We should move the name check to line 114, so we don't need search for ancestors when they are not keywords.